js-exercises/drop-elements/dropElements.js

```diff
@@ -0,0 +1,14 @@
+function dropElements(elements, predicate) {
+  const arr = elements.map(element => element);
+  const newArr = arr.filter((element) => {
+    const shouldNotDropElement = predicate(element);
+    if (shouldNotDropElement) {
+      return true;
+    }
+    return false;
+  });
+  console.log(newArr);
+  return newArr;
+}
+
+export { dropElements };
```

Comment:
1. I guess you used **elements.map** to remove the holes of an array. But since the filter method doesn't iterate the holes, **elements.map** becomes redundant.
2. Naming could be little better ( eg: arr, newArr)

Reply:
Hi, thanks for reviewing. The rules of functional programming is, you should never mutate the original parameters, so I am storing the array element into a new variable without referencing.
A map is also used to deep copy an array.
What is your opinion on this?

Reply:
I will use a more generic name, thanks for pointing.